From 1c0ec4c1a97924e6399df9a349551878ebcfa102 Mon Sep 17 00:00:00 2001
From: Mian Yousaf Kaukab <ykaukab@suse.de>
Date: Tue, 24 Sep 2019 17:10:24 +0200
Subject: [PATCH 1/1] Revert "ARM: tegra: reserve unmapped RAM so EFI doesn't
 use it"

This reverts commit 0797f7f0b7e1d7853e2842ddc235ffef139fa792.

Tegra specific solution is not required any more as efi core has been
made aware of ram_top with the following commit:
7b78d6438a efi_loader: Reserve unaccessible memory

Signed-off-by: Mian Yousaf Kaukab <ykaukab@suse.de>
---
 arch/arm/mach-tegra/board2.c | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/arch/arm/mach-tegra/board2.c b/arch/arm/mach-tegra/board2.c
index f13bd256cc..07f54f0684 100644
--- a/arch/arm/mach-tegra/board2.c
+++ b/arch/arm/mach-tegra/board2.c
@@ -6,7 +6,6 @@
 
 #include <common.h>
 #include <dm.h>
-#include <efi_loader.h>
 #include <env.h>
 #include <errno.h>
 #include <ns16550.h>
@@ -224,19 +223,6 @@ int board_early_init_f(void)
 
 int board_late_init(void)
 {
-#if CONFIG_IS_ENABLED(EFI_LOADER)
-	if (gd->bd->bi_dram[1].start) {
-		/*
-		 * Only bank 0 is below board_get_usable_ram_top(), so all of
-		 * bank 1 is not mapped by the U-Boot MMU configuration, and so
-		 * we must prevent EFI from using it.
-		 */
-		efi_add_memory_map(gd->bd->bi_dram[1].start,
-				   gd->bd->bi_dram[1].size >> EFI_PAGE_SHIFT,
-				   EFI_BOOT_SERVICES_DATA, false);
-	}
-#endif
-
 #if defined(CONFIG_TEGRA_SUPPORT_NON_SECURE)
 	if (tegra_cpu_is_non_secure()) {
 		printf("CPU is in NS mode\n");
-- 
2.20.1

